src/Ethereum/ABI/Tuple.h

```diff
@@ -89,25 +89,70 @@ void encode_dynamic(const std::tuple<First, T...>& tuple, Data& data) {
 }
 
 template <typename... T>
-void encode(const std::tuple<T...>& tuple, Data& data) {
+inline void encode(const std::tuple<T...>& tuple, Data& data) {
     std::size_t headSize = head_size(tuple);
 
     std::size_t dynamicOffset = 0;
     encode_small(tuple, headSize, dynamicOffset, data);
     encode_dynamic(tuple, data);
 }
 
-std::string type_string(const std::tuple<>& tuple) {
+inline std::string type_string(const std::tuple<>& tuple) {
     return "";
 }
 
 template <typename... T>
-std::string type_string(const std::tuple<T...>& tuple) {
+inline std::string type_string(const std::tuple<T...>& tuple) {
     std::string string = type_string(head(tuple));
     if (std::tuple_size<std::tuple<T...>>::value > 1) {
         string += "," + type_string(tail(tuple));
     }
     return string;
 }
 
+inline bool decodeTuple_small(const Data& encoded, std::tuple<>& tuple_inout, size_t& offset_inout) { return true; }
+
+/// Process small values and extract indices for larger values
+template <typename First, typename... T>
+inline bool decodeTuple_small(const Data& encoded, std::tuple<First, T...>& tuple_inout, size_t& offset_inout) {
```

Comment:
just naming, `decodeTupleSmall` and `decodeTupleLarge`

Reply:
ok,accepted, thx